strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/event/artifact/MavenArtifactDownloadedEventListener.java

```diff
@@ -0,0 +1,147 @@
+import org.carlspring.strongbox.client.ArtifactTransportException;
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.providers.ProviderImplementationException;
+import org.carlspring.strongbox.providers.layout.LayoutProvider;
+import org.carlspring.strongbox.providers.layout.LayoutProviderRegistry;
+import org.carlspring.strongbox.providers.repository.proxied.LocalStorageProxyRepositoryArtifactResolver;
+import org.carlspring.strongbox.providers.repository.proxied.ProxyRepositoryArtifactResolver;
+import org.carlspring.strongbox.providers.repository.proxied.SimpleProxyRepositoryArtifactResolver;
+import org.carlspring.strongbox.services.ArtifactMetadataService;
+import org.carlspring.strongbox.services.ArtifactResolutionService;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryLayoutEnum;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.NoSuchFileException;
+import java.security.NoSuchAlgorithmException;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.apache.maven.index.locator.Locator;
+import org.apache.maven.index.locator.MetadataLocator;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class MavenArtifactDownloadedEventListener
+        implements ArtifactEventListener
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(MavenArtifactDownloadedEventListener.class);
+
+    @Inject
+    private ConfigurationManager configurationManager;
+
+    @Inject
+    private ArtifactResolutionService artifactResolutionService;
+
+    @Inject
+    private ArtifactMetadataService artifactMetadataService;
+
+    @Inject
+    @SimpleProxyRepositoryArtifactResolver.SimpleProxyRepositoryArtifactResolverQualifier
+    private ProxyRepositoryArtifactResolver simpleProxyRepositoryArtifactResolver;
+
+    @Inject
+    @LocalStorageProxyRepositoryArtifactResolver.LocalStorageProxyRepositoryArtifactResolverQualifier
+    private ProxyRepositoryArtifactResolver localStorageProxyRepositoryArtifactResolver;
+
+    @Inject
+    private LayoutProviderRegistry layoutProviderRegistry;
+
+    private Locator metadataLocator = new MetadataLocator();
+
+    @Override
+    public void handle(ArtifactEvent event)
+    {
+        if (event.getType() != ArtifactEventTypeEnum.EVENT_ARTIFACT_FILE_DOWNLOADED.getType())
+        {
+            return;
+        }
+
+        Repository repository = getRepository(event);
+
+        if (!RepositoryLayoutEnum.MAVEN_2.getLayout().equals(repository.getLayout()))
```

Comment:
This should actually probably be the first check of the method...

Reply:
No problem, whatever you like.